codetools/contexts/tests/multi_context_test_case.py

```diff
@@ -1,23 +1,36 @@
 # Standard library imports
 from io import BytesIO
-import os
-import sys
-import unittest
-
-import nose
-from nose.plugins.skip import SkipTest
 
 # Enthought library imports
+from traits.adaptation.api import (
+    AdaptationManager,
+    get_global_adaptation_manager,
+    set_global_adaptation_manager,
+)
 from traits.api import Any
 
 # Geo library imports
-from codetools.contexts.tests.abstract_context_test_case import AbstractContextTestCase
-from codetools.contexts.data_context import DataContext
+from codetools.contexts.data_context import (
+    DataContext, register_i_context_adapter_offers)
+from codetools.contexts.i_context import (
+    register_dict_to_context_adapter_offers)
 from codetools.contexts.multi_context import MultiContext
+from codetools.contexts.tests.abstract_context_test_case import AbstractContextTestCase
 
 
 class MultiContextTestCase(AbstractContextTestCase):
 
+    def setUp(self):
```

Comment:
Hmm. My initial thought was that I wasn't able to use the `TestCaseWithAdaptation` base class here - the test class hierarchy gets too messy, since `AbstractContextTestCase` _also_ inherits from `TestCase`. But it looks as though there's no overlap between the methods, so we may just be able to use multiple inheritance here. I'll try that.

Reply:
Done.